[RELEASE] Release version 1.2.2

## Release OpenSearch and OpenSearch Dashboards 1.2.2

This is an out of band release, it will follow the release process but normal steps will be abbreviated.

See https://opensearch.org/blog/releases/2021/12/update-to-1-2-1/ for more information.

### Preparation

- [x] Assign this issue to a release owner.
- [x] Document any new quality requirements or changes.
- [x] [Create a version label](https://github.com/opensearch-project/opensearch-plugins/blob/main/META.md#create-or-update-labels-in-all-plugin-repos) in each component repo for this, and the next minor release.
- [x] Confirm that OpenSearch-Dashboards does not need an updated manifest

### Development

- [x] OpenSearch: opensearch-project/OpenSearch#1728
- [x] security https://github.com/opensearch-project/opensearch-build/pull/1339
- [x] common-utils https://github.com/opensearch-project/common-utils/pull/104
- [x] job-scheduler https://github.com/opensearch-project/job-scheduler/pull/109
- [x] alerting https://github.com/opensearch-project/alerting/pull/257
- [x] asynchronous-search https://github.com/opensearch-project/asynchronous-search/pull/79
- [x] index-management https://github.com/opensearch-project/index-management/pull/225
- [x] k-NN https://github.com/opensearch-project/k-NN/pull/243
- [x] security https://github.com/opensearch-project/security/pull/1525
- [x] performance-analyzer https://github.com/opensearch-project/performance-analyzer/pull/103
- [x] anomaly-detection https://github.com/opensearch-project/anomaly-detection/pull/344
- [x] cross-cluster-replication https://github.com/opensearch-project/cross-cluster-replication/pull/269
- [x] sql https://github.com/opensearch-project/sql/pull/337
- [x] dashboards-reports https://github.com/opensearch-project/dashboards-reports/pull/255
- [x] observability https://github.com/opensearch-project/observability/pull/346
- [x] Make pull requests to add each component to [manifests/1.2.2/opensearch-1.2.2.yml](/opensearch-project/opensearch-build/tree/main/manifests/1.2.2/opensearch-1.2.2.yml) https://github.com/opensearch-project/opensearch-build/pull/1345
- [x] All components have been updated
- [x] Declare a release candidate build
- [x] Create an issue publish all test results.  (+1.5 hour)
  ~- [ ] Performance tests, tracked on this issue~
  - [x] Integration tests, tracked on this issue
- [ ] Verify all issued labeled `v1.2.2` in all projects have been resolved.
- [x] #1356
- [x] Complete [documentation](https://github.com/opensearch-project/documentation-website) for this release.
- [x] Maven release
- [x] Native plugins release
- [x] Publish Docker images
- [x] Author [blog post](https://github.com/opensearch-project/project-website) for this release.
- [x] Gather, review and publish release notes. [git-release-notes](https://github.com/ariatemplates/git-release-notes) may be used to generate release notes from your commit history.
- [x] Publish this release on [opensearch.org](https://opensearch.org/downloads.html).
- [x] Publish forum posts - release is launched!
- [x] Publish [blog post](https://github.com/opensearch-project/project-website) - release is launched!

### Post Release

- [x] Create [release tags](https://github.com/opensearch-project/.github/blob/main/RELEASING.md#tagging) for each component.
- [x] Replace refs in [manifests/1.2.2](/opensearch-project/opensearch-build/tree/main/manifests/1.2.2) with tags.
- [x] Update [this template](https://github.com/opensearch-project/opensearch-build/blob/main/.github/ISSUE_TEMPLATE/release_template.md) with any new or missed steps.
- [x] Conduct a retrospective, and publish its results. https://github.com/opensearch-project/opensearch-build/issues/1360

Faced the below error while running integration testing:
```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':alerting:integTest'.
> Could not resolve all files for configuration ':alerting:opensearch_distro_extracted_testclusters-alerting-integTest-0-1.2.2-'.
   > Could not resolve opensearch-distribution:opensearch:1.2.2.
     Required by:
         project :alerting
      > Could not resolve opensearch-distribution:opensearch:1.2.2.
         > Could not get resource 'https://artifacts.opensearch.org/releases/core/opensearch/1.2.2/opensearch-min-1.2.2-linux-x64.tar.gz'.
            > Could not HEAD 'https://artifacts.opensearch.org/releases/core/opensearch/1.2.2/opensearch-min-1.2.2-linux-x64.tar.gz'. Received status code 403 from server: Forbidden
      > Could not resolve opensearch-distribution:opensearch:1.2.2.
         > Could not get resource 'https://artifacts.opensearch.org/release-candidates/core/opensearch/1.2.2/opensearch-min-1.2.2-linux-x64.tar.gz'.
            > Could not HEAD 'https://artifacts.opensearch.org/release-candidates/core/opensearch/1.2.2/opensearch-min-1.2.2-linux-x64.tar.gz'. Received status code 403 from server: Forbidden
```
I suspect it is related to https://github.com/opensearch-project/OpenSearch/issues/1240
@peterzhuamazon is working on running the snapshot workflow that will place the artifact in that location.

--- a/manifests/1.2.2/opensearch-1.2.2.yml
+++ b/manifests/1.2.2/opensearch-1.2.2.yml
@@ -10,43 +10,43 @@ build:
 components:
   - name: OpenSearch
     repository: https://github.com/opensearch-project/OpenSearch.git
-    ref: "1.2"
+    ref: tags/1.2.2
     checks:
       - gradle:publish
       - gradle:properties:version
   - name: common-utils
     repository: https://github.com/opensearch-project/common-utils.git
-    ref: "1.2"
+    ref: tags/1.2.2.0
     checks:
       - gradle:publish
       - gradle:properties:version
   - name: job-scheduler
     repository: https://github.com/opensearch-project/job-scheduler.git
-    ref: "1.2"
+    ref: tags/1.2.2.0
     checks:
       - gradle:properties:version
       - gradle:dependencies:opensearch.version
   - name: alerting
     repository: https://github.com/opensearch-project/alerting.git
-    ref: "1.2"
+    ref: tags/1.2.2.0
     checks:
       - gradle:properties:version
       - gradle:dependencies:opensearch.version: alerting
   - name: asynchronous-search
     repository: https://github.com/opensearch-project/asynchronous-search.git
-    ref: "1.2"
+    ref: tags/1.2.2.0
     checks:
       - gradle:properties:version
       - gradle:dependencies:opensearch.version
   - name: index-management
     repository: https://github.com/opensearch-project/index-management.git
-    ref: "1.2"
+    ref: tags/1.2.2.0
     checks:
       - gradle:properties:version
       - gradle:dependencies:opensearch.version
   - name: k-NN
     repository: https://github.com/opensearch-project/k-NN.git
-    ref: "1.2"
+    ref: tags/1.2.2.0
     platforms:
       - darwin
       - linux
@@ -55,10 +55,10 @@ components:
       - gradle:dependencies:opensearch.version
   - name: security
     repository: https://github.com/opensearch-project/security.git
-    ref: "1.2"
+    ref: tags/1.2.2.0
   - name: performance-analyzer
     repository: https://github.com/opensearch-project/performance-analyzer.git
-    ref: "1.2"
+    ref: tags/1.2.2.0
     checks:
       - gradle:properties:version
       - gradle:dependencies:opensearch.version
@@ -67,32 +67,32 @@ components:
       - linux
   - name: anomaly-detection
     repository: https://github.com/opensearch-project/anomaly-detection.git
-    ref: "1.2"
+    ref: tags/1.2.2.0
     checks:
       - gradle:properties:version
       - gradle:dependencies:opensearch.version
   - name: cross-cluster-replication
     repository: https://github.com/opensearch-project/cross-cluster-replication.git
-    ref: "1.2"
+    ref: tags/1.2.2.0
     checks:
       - gradle:properties:version
       - gradle:dependencies:opensearch.version
   - name: sql
     repository: https://github.com/opensearch-project/sql.git
-    ref: "1.2"
+    ref: tags/1.2.2.0
     checks:
       - gradle:properties:version
       - gradle:dependencies:opensearch.version: plugin
   - name: dashboards-reports
     repository: https://github.com/opensearch-project/dashboards-reports.git
-    ref: "1.2"
+    ref: tags/1.2.2.0
     working_directory: reports-scheduler
     checks:
       - gradle:properties:version
       - gradle:dependencies:opensearch.version
   - name: opensearch-observability
     repository: https://github.com/opensearch-project/observability
-    ref: "1.2"
+    ref: tags/1.2.2.0
     working_directory: opensearch-observability
     checks:
       - gradle:properties:version

